codebasin/__init__.py

```diff
@@ -123,3 +127,112 @@ def from_json(cls, instance: dict):
             command=command,
             output=output,
         )
+
+
+class CodeBase:
+    """
+    A representation of all source files in the code base.
+
+    Attributes
+    ----------
+    directories: list[str | os.PathLike[str]]
+        The set of source directories that make up the code base.
+
+    exclude_patterns: list[str]
+        A set of patterns describing source files excluded from the code base.
+    """
+
+    def __init__(
+        self,
+        *directories: str | os.PathLike[str],
```

Comment:
Very minor thing: semantically I get `os.PathLike`, but I'm struggling to differentiate between `str` and `os.PathLike[str]`. Normally, I opt for `str | Path`, but in this case would it make more sense to use `os.PathLike[str] | Path`?

Reply:
I wasn't sure, but when I looked into this, a `str` doesn't work with `PathLike[str]`.  `str | PathLike[str]` is a pattern I saw in `typeshed` and other places.